### ISSUE TITLE
"unused variable" warning types needing discussion before they're fixed

### ISSUE THREAD

**Issue by akien-mga** [godotengine/godot#5568]
When compiling Godot with GCC 5.4.0, we get tons of `[-Wunused-variable]` and `[-Wunused-but-set-variable]` warnings for this kind of things:

```
drivers/chibi/cp_loader_s3m.cpp: In member function 'CPLoader::Error CPLoader_S3M::load_sample(CPSample*)':
drivers/chibi/cp_loader_s3m.cpp:147:7: warning: unused variable 'dsk' [-Wunused-variable]
   int dsk=file->get_byte();
       ^
drivers/chibi/cp_loader_s3m.cpp:148:7: warning: unused variable 'pack' [-Wunused-variable]
   int pack=file->get_byte();
       ^
```

In our various loaders in `drivers/`, we quite often use `int some_var_that_will_never_be_used = file->get_byte();` just to move further while decoding the file.

It's not a big deal, but getting rid of these warnings would be nice. What would be the good practice in such a case? Just invoke `file->get_byte(); // pack` without assigning its return value?

Suggestions welcome, then I can handle the fix for all occurrences (there are quite a few).

**Comment by akien-mga** [comment]
Another related example, which is an unused variable used to receive the return value of a method that does stuff:
```
drivers/theora/video_stream_theora.cpp: In member function 'virtual void VideoStreamPlaybackTheora::update(float)':
drivers/theora/video_stream_theora.cpp:605:9: warning: unused variable 'tr' [-Wunused-variable]
     int tr = vorbis_synthesis_read(&vd, ret-to_read);
         ^
```

In such a case, should we also just call `vorbis_synthesis_read(&vd, ret-to_read);` without caring about the return value, or should we actually do something with this value?

**Comment by vnen** [comment]
I guess this might be view case by case. If a function is used just to advance the cursor then it might be just ignored. But the Vorbis function returns this (according to [reference](https://xiph.org/vorbis/doc/libvorbis/vorbis_synthesis_read.html)):

> * 0 on success.
* OV_EINVAL if samples is greater than the number of remaining samples in the buffer.

This may deserve some error checking conditional, in which the variable would be used.

**Comment by akien-mga** [comment]
Another situation where I'm not 100% sure what would be the best practice: https://github.com/godotengine/godot/blob/master/platform/android/export/export.cpp#L1357-L1362

**Comment by akien-mga** [comment]
Another weird one (which is present in several places in `node.cpp` and also in `scene_main_loop.cpp`: https://github.com/godotengine/godot/blob/17e4ead/scene/main/node.cpp#L53-L54
Are those variables used implicitly by the following function call, or just useless?

**Comment by bojidar-bg** [comment]
@akien-mga Well, that one looks useless, except if the `CallError` constructor does something modifying the global state, since non-pointers are local to the place of execution (or should be until something very weird starts going on).

**Comment by akien-mga** [comment]
Another one I'm not sure about: https://github.com/godotengine/godot/blob/bccdc11/core/translation.cpp#L703 (just dropping that here to keep track of it)

**Comment by paper-pauper** [comment]
My 2c on it: if a variable is not never used, it should not be declared at all. But in some of those cases a comment (which does not generate any warnings) would solve the issue, as long as the same "mark" or "tag" is used.

Something like 

```c++
 OS::get_singleton()->execute(jarsigner,args,true,NULL,NULL,&retval); // [return value discarded]
```

so that you can easily find those via grep, etc.

**Comment by akien-mga** [comment]
Fixed by #10908.
